ui/tests/e2e/fixtures/environmentsPage.js

```diff
@@ -0,0 +1,253 @@
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+  }
+  async fillInput(locator, value) {
+    await locator.fill(value);
+  }
+
+  async goToProfiles() {
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+  }
+
+  async goToConnections() {
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/connections`);
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/connections`);
+  }
+
+  async doesProfileExist(environmentName) {
+    await this.goToProfiles();
+    return this.page
+      .locator(`text=${environmentName}`)
+      .count()
+      .then((count) => count === 1);
+  }
+  async createEnvironmentProfile(environmentName) {
+    if (!(await this.doesProfileExist(environmentName))) {
+      await this.page.getByRole('button', { name: 'Create' }).click();
+      await this.fillInput(this.page.locator('#Environment_name'), `${environmentName}`);
+      await this.fillInput(
+        this.page.locator('#Environment_description'),
+        'Created profile for playwright tests',
+      );
+      await this.page.getByRole('button', { name: 'Save', exact: true }).click();
+      // Check for notification visibility
+      const notification = await this.page.getByTestId('notification-success-bar');
+      await expect(notification).toBeVisible();
+      await expect(notification).toBeHidden({ timeout: 30000 });
+
+      // Wait for the profile card to appear
+      await this.page.waitForSelector(`text=${environmentName}`, {
+        state: 'visible',
+        timeout: 3 * 60 * 1000,
+      });
+
+      // Additional check: wait for the profile card to be clickable
+      await this.page.waitForSelector(`text=${environmentName}`, {
+        state: 'visible',
+        timeout: 30000,
+      });
+
+      console.log(`Profile "${environmentName}" has been created and is visible.`);
+    } else {
+      console.log(`Profile "${environmentName}" already exists.`);
+    }
+  }
+
+  async assignConnections(environmentName) {
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+    await this.page.waitForSelector(`text=${environmentName}`, { state: 'visible' });
+
+    // Click the button using the newly added data-testid
+    await this.page.getByTestId(`assigned-conn-${environmentName}`).click();
+
+    // Wait for and locate the dialog
+    const dialog = await this.page.locator('[aria-labelledby="alert-dialog-slide-title"]');
+    await expect(dialog).toBeVisible();
+
+    // Define the aria-labelledby values for checkboxes
+    const checkBoxAriaLabels = [
+      'transfer-list-item-kubernetes-mantra-9-label',
+      'transfer-list-item-meshery-themis-7-label',
+      'transfer-list-item-meshery-vader-9-label',
+      'transfer-list-item-meshery-exogorth-0-label',
+      'transfer-list-item-meshery-hydra-8-label',
+    ];
```

Comment:
Are these resources always available, there are no rotations in terms of naming right?

Reply:
the connections may not always be available. I'll change it to choose connections that are available.